eec-backend/Controllers/ProductController.cs

```
@@ -0,0 +1,100 @@
+﻿using eec_backend.Data;
+using eec_backend.Models;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+
+namespace eec_backend.Controllers
+{
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ProductController : ControllerBase
+    {
+        private readonly ProductContext _context;
+
+        public ProductController(ProductContext context)
        {
+            _context = context;
+        }
+
+        // GET: api/Products
+        [HttpGet]
+        public async Task<ActionResult<IEnumerable<Product>>> GetProducts()
+        {
+            return await _context.Products.ToListAsync();
+        }
+
+        // GET: api/Products/{modelIdentifier}
+        [HttpGet("{modelIdentifier}")]
+        public async Task<ActionResult<Product>> GetProduct(string modelIdentifier)
+        {
+            var product = await _context.Products.FindAsync(modelIdentifier);
+
+            if (product == null)
+            {
+                return NotFound();
+            }
+
+            return product;
+        }
+
+        // PUT: api/Products/{modelIdentifier}
+        [HttpPut("{modelIdentifier}")]
+        public async Task<IActionResult> PutProduct(string modelIdentifier, Product product)
+        {
+            if (modelIdentifier != product.ModelIdentifier)
+            {
+                return BadRequest();
+            }
+
+            _context.Entry(product).State = EntityState.Modified;
+
+            try
+            {
+                await _context.SaveChangesAsync();
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                if (!ProductExists(modelIdentifier))

```

Please dont throw errors. We should rather return status 500 with exception message to not block the application running. Also move this if statement higher so we can fail faster if no product exists.

---

`return StatusCode(500, ex.Message)` and add `ex` after `DbUpdateConcurrencyException`

---

We could also use logger for logging such a details inside our app but this is not really important right now. Maybe I will do it later in some of my PR